lib/Sema/CSSimplify.cpp

```diff
@@ -7638,6 +7648,16 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
     return SolutionKind::Solved;
   }
 
+  case FixKind::ExpandArrayIntoVarargs: {
+    auto result = matchTypes(type1, ArraySliceType::get(type2), matchKind,
```

Comment:
I think this matching should be moved to `attempt` and you'd want to do a reverse comparison vs. what it is right now - get a type from `isArrayType` and match it against `type` with `TMF_ApplyingFix` flag set.

Reply:
I was able to reverse the comparison and add the `TMF_ApplyingFixFlag`, I'm not sure the type matching can be moved into `repairFailures`/`attempt` though. When I tried, it failed to diagnose array literals properly because their element types were still type variables.

Reply:
Do you know what exactly when wrong? I'd expect conversion from type variable associated with element type to be successful. It kind of sounds like the fix has to be delayed for some cases...

Reply:
Oh, I think I figured it out. The type variable did not have a fixed type when `matchTypes` was called, and because TMF_GenerateConstraints wasn't passed it gave up immediately after. I added the generate constraints flag and everything works now. (aside from[SR-11104](https://bugs.swift.org/browse/SR-11104), which is a known issue)

Reply:
Makes sense, ideally we would want to avoid creating new constraints like that because they can fail later (which invalidates the fix), but we can't really do much about that at this point...